manifests/install.pp

```diff
@@ -78,4 +82,11 @@
       system => true,
     })
   }
+  file { $prometheus::config_dir:
```

Comment:
as I mentioned in a comment above: this was necessary to add the missing dependency between the config file and the alerts file without creating a dependency cycle. Now both depend on the directory, which gets created first during install.
The config file needs the alerts file, because the newly introduced validation with `promtool` chokes if it can't find the named alerts file.
That's also the reason why I removed the conditional creation of the alerts file and create it always (even if empty). Otherwise the dependency would have to be conditional on the alert file's creation and the `prometheus.yaml.erb` template would also have to be changed to not always reference a possibly non-existent alerts file.
In summary: it just felt like the path of least resistance :wink:

Reply:
ACK +1